glslc/test/include.py

```diff
@@ -306,3 +306,203 @@ class TestWrongPoundVersionInIncludedFile(expect.ErrorMessage):
         "b.glsl:1: error: '#version' : must occur first in shader\n",
         '1 error generated.\n'
     ]
+
+
+@inside_glslc_testsuite('Include')
+class VerifyRelativeInclude(expect.StdoutMatch):
+    """Tests #including a relative sibling."""
+
+    environment = Directory('.', [
+        File('a.vert', '#version 140\ncontent a\n#include "foo/b.glsl"\n'),
+        Directory('foo', [
+            File('b.glsl', '#include "c.glsl"\ncontent b\n'),
+            File('c.glsl', 'content c\n')
+        ])])
+
+    glslc_args = ['-E', 'a.vert']
+
+    expected_stdout = \
+"""#version 140
+#extension GL_GOOGLE_include_directive : enable
+#line 0 "a.vert"
+
+content a
+#line 0 "foo/b.glsl"
+#line 0 "foo/c.glsl"
+ content c
+#line 1 "foo/b.glsl"
+ content b
+#line 3 "a.vert"
+
+"""
+
+@inside_glslc_testsuite('Include')
+class VerifyNestedRelativeInclude(expect.StdoutMatch):
+    """Tests #including a relative child file."""
+
+    environment = Directory('.', [
+        File('a.vert', '#version 140\ncontent a\n#include "foo/b.glsl"\n'),
+        Directory('foo', [
+            File('b.glsl', '#include "bar/c.glsl"\ncontent b\n'),
+            Directory('bar', [
+                File('c.glsl', 'content c\n')
+            ])
+        ])
+    ])
+
+    glslc_args = ['-E', 'a.vert']
+
+    expected_stdout = \
+"""#version 140
+#extension GL_GOOGLE_include_directive : enable
+#line 0 "a.vert"
+
+content a
+#line 0 "foo/b.glsl"
+#line 0 "foo/bar/c.glsl"
+ content c
+#line 1 "foo/b.glsl"
+ content b
+#line 3 "a.vert"
+
+"""
+
+
+@inside_glslc_testsuite('Include')
+class VerifyRelativeIncludeWithDashI(expect.StdoutMatch):
+    """Tests #including a relative file from a -I directory."""
+
+    environment = Directory('.', [
+        File('a.vert', '#version 140\ncontent a\n#include "b.glsl"\n'),
+        Directory('foo', [
+            File('b.glsl', '#include "c.glsl"\ncontent b\n'),
```

Comment:
This was intended to test the "b.glsl" include, but I have made the change, as it makes the test more robust.